plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -609,13 +609,16 @@ protected String getColumnDefinitionSql(ConnectorSession session, ColumnMetadata
     @Override
     public JdbcOutputTableHandle beginInsertTable(ConnectorSession session, JdbcTableHandle tableHandle, List<JdbcColumnHandle> columns)
     {
+        RemoteTableName remoteTableName = tableHandle.asPlainTable().getRemoteTableName();
         SchemaTableName schemaTableName = tableHandle.asPlainTable().getSchemaTableName();
         ConnectorIdentity identity = session.getIdentity();
 
         try (Connection connection = connectionFactory.openConnection(session)) {
-            String remoteSchema = identifierMapping.toRemoteSchemaName(identity, connection, schemaTableName.getSchemaName());
-            String remoteTable = identifierMapping.toRemoteTableName(identity, connection, remoteSchema, schemaTableName.getTableName());
-            String catalog = connection.getCatalog();
+            String catalog = remoteTableName.getCatalogName()
+                    .orElse(connection.getCatalog());
+            String remoteSchema = remoteTableName.getSchemaName()
+                    .orElse(identifierMapping.toRemoteSchemaName(identity, connection, schemaTableName.getSchemaName()));
```

Comment:
Why do we need an .orElse here?

(also, nit: `orElseGet` is your friend)

Reply:
In case of `null` InteliJ recommends `orElse`

Reply:
> Why do we need an .orElse here?

In some cases `RemoteSchema` might be null - so we will be adding it as a fallback. But I guess it makes sense if we pass `RemoteTable` as a part of `JdbcOutputTableHandle`. WDYT ?